Xabe.FFmpeg/Streams/WebStream.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace Xabe.FFmpeg.Streams
+{
+    /// <summary>
+    ///     Stream with foreign web source
+    /// </summary>
+    public class WebStream : IStream
+    {
+        private readonly Uri _uri;
+        private readonly TimeSpan? _duration;
+
+        /// <summary>
+        /// Create web stream
+        /// </summary>
+        /// <param name="uri">File uri</param>
+        /// <param name="format">Stream format</param>
+        /// <param name="duration">Duration of video</param>
+        public WebStream(Uri uri, string format, TimeSpan? duration)
```

Comment:
conversion will be stoppable in next work item so I will create test for null timespan